src/components/Search/index.js

```diff
@@ -2,12 +2,16 @@ import React from 'react'
 
 import TokenSearch from './TokenSearch'
 import NFTSearch from './NFTSearch'
-import { useNFTApp } from '../../hooks'
+import YieldSearch from './YieldSearch'
+import { useNFTApp, useYieldApp } from '../../hooks'
 
 export default function Search({ ...props }) {
   const isNFTApp = useNFTApp()
+  const isYieldApp = useYieldApp()
   if (isNFTApp) {
-    return <NFTSearch {...props}/>
+    return <NFTSearch {...props} />
+  } else if (isYieldApp) {
```

Comment:
looks like this is very close to the current search, would be good to avoid repeating code but if it's hard to make it that way then imo no problem we can just repeat it

Reply:
good one, will modifiy

Reply:
deleted YieldSearch and made some changes in TokenSearch. I ran locally and from the few manual tests everything seemed to work fine. lmk if u guys see anything weird here, especially this at the top (though i guess its fine):
```
const OpenTokenSearch = dynamic(() => import('./OpenTokenSearch'))
const OpenYieldSearch = dynamic(() => import('./OpenYieldSearch'))
```